service/controller/app/v1/resource/configmap/create.go

```diff
@@ -28,6 +32,24 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
 		_, err = cc.K8sClient.CoreV1().ConfigMaps(configMap.Namespace).Create(configMap)
 		if apierrors.IsAlreadyExists(err) {
 			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("already created configmap %#q in namespace %#q", configMap.Name, configMap.Namespace))
+		} else if apierrors.IsTimeout(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "cluster api timeout.")
+
+			// We should not hammer API if it is not available, the cluster
+			// might be initializing. We will retry on next reconciliation loop.
+			resourcecanceledcontext.SetCanceled(ctx)
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+
+			return nil
+		} else if !key.InCluster(cr) && tenant.IsAPINotAvailable(err) {
```

Comment:
I would check first whether it's InCluster or not.

Reply:
I do not understand this condition. Why `!key.InCluster(cr)`? Which problem does this solve?

Reply:
Shit, will cover this.